cmd/participation/join.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"os"
+)
+
+// Joins a OSN to a new or existing channel.
+func join(osn, tlsDir, channelID, configBlockPath string) {
+	blockBytes, err := ioutil.ReadFile(configBlockPath)
+	if err != nil {
+		panic(err)
+	}
+
+	// NOTE: this is in its own function because we technically
+	// don't need it. we also don't need to unmarshal the block bytes
+	// if we don't care to validate the channel id in the block
+	err = validateBlockChannelID(blockBytes, channelID)
+	if err != nil {
+		panic(err)
+	}
+
+	url := fmt.Sprintf("https://%s/participation/v1/channels", osn)
+
+	joinBody := new(bytes.Buffer)
+	writer := multipart.NewWriter(joinBody)
+	part, err := writer.CreateFormFile("config-block", fmt.Sprintf("%s.block", channelID))
+	if err != nil {
+		panic(err)
+	}
+	part.Write(blockBytes)
+	err = writer.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	req, err := http.NewRequest(http.MethodPost, url, joinBody)
+	if err != nil {
+		panic(err)
+	}
+	req.Header.Set("Content-Type", writer.FormDataContentType())
+	client := httpClient(tlsDir)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		panic(err)
+	}
+	resp.Body.Close()
+
+	err = printResponseAsJSON(bodyBytes, &channelInfo{}, os.Stdout)
```

Comment:
In general, it's a little odd to have the console printing pushed out into the leaf of the calls.  I'd generally expect for a response and error to be propagated back up to the top, where, uniform decisions can be made about how to report success and failure, the output format, etc.

Reply:
Refactored. Done.